src/value/owned.rs

```diff
@@ -272,13 +272,20 @@ impl Struct for OwnedStruct {
 impl PartialEq for OwnedStruct {
     fn eq(&self, other: &Self) -> bool {
         for (key, value) in &self.text_fields {
-            if other.get_all(key).ne(value.iter().map(|(_s, v)| v)) {
-                return false;
-            }
-        }
-        if other.no_text_fields.ne(&self.no_text_fields) {
-            return false;
+            value.iter().all(|(_my_s, my_v)| {
+                other
+                    .get_all(key)
+                    .find(|other_v| my_v == *other_v)
+                    .is_some()
+            });
         }
+        self.no_text_fields.iter().all(|(my_k, my_v)| {
+            other
+                .no_text_fields
+                .iter()
+                .find(|(other_k, other_v)| my_k == other_k && my_v == other_v)
+                .is_some()
+        });
         true
```

Comment:
I think this is written with side effects being ignored, so I *think* this always returns true.  In other words, I don't think you're doing anything with the predicate and I would like to see some negative cases being tested (e.g. `{a:1} != {a:2}`.  The end statement `true` is a tell.

Consider rewriting as a single expression instead of with the `for` loop, because I think that is getting in your way, but also add some struct cases that deal with non-equivalence.

For example, something like:
```rust
self.text_fields.all(|(key, values)| {
  values.iter().all(...)
}) && self.no_text_fields.iter().all(...)
```

Would eliminate the need for imperative branching (though I do think it hurts readability for readers not used to functional style, so YMMV).